crates/core/src/context.rs

```diff
@@ -10,25 +11,34 @@ use crate::{
     problem::FileOwners,
 };
 use anyhow::Result;
+use grit_util::AstNode;
 use marzano_language::target_language::TargetLanguage;
 use marzano_util::analysis_logs::AnalysisLogs;
 
-pub trait Context {
-    fn pattern_definitions(&self) -> &[PatternDefinition];
+/// Contains various kinds of context about the problem being executed.
+pub trait ProblemContext: Clone + std::fmt::Debug + Sized {
```

Comment:
I'm trying to phase out the "Problem" nomenclature. Let's call this `QueryContext`.